addons/payment_razorpay/models/payment_provider.py

```diff
@@ -76,12 +95,18 @@ def _razorpay_make_request(self, endpoint, payload=None, method='POST'):
         self.ensure_one()
 
         url = url_join('https://api.razorpay.com/v1/', endpoint)
-        auth = (self.razorpay_key_id, self.razorpay_key_secret)
+        headers = {}
+        auth = None
+        # razorpay_access_token = self._get_razorpay_access_token()
+        if self.razorpay_access_token:
+            headers = {'Authorization': f'Bearer {self.razorpay_access_token}'}
+        else:
+            auth = (self.razorpay_key_id, self.razorpay_key_secret)
```

Comment:
```suggestion
     
         razorpay_headers = {}
         if self.razorpay_access_token:
                headers['Authorization'] = f'Bearer {self.razorpay_access_token}'

        razorpay_auth = None
        if not self.razorpay_access_token:
              auth = (self.razorpay_key_id, self.razorpay_key_secret)
```
Easier to understand